src/extra/gpkg.jl

```diff
@@ -0,0 +1,1088 @@
+    byte_order_inner = read(io, UInt8)
+    read_value_inner = byte_order_inner == 0x01 ? ltoh : ntoh
+    wkb_type_inner = read_value_inner(read(io, UInt32))
+    
+    # Parse the individual geometry's Z/M flags from its own header
+    inner_base_type = wkb_type_inner & 0x0FFFFFFF
+    inner_has_z = (wkb_type_inner & WKB_Z) != 0
+    inner_has_m = (wkb_type_inner & WKB_M) != 0
+    
+    poly = parse_wkb_polygon(io, byte_order_inner, inner_has_z, inner_has_m, crs_type)
+    if !isnothing(poly)
+      push!(polygons, poly)
+    end
+  end
+  
+  # Create Multi geometry with fallback to avoid StackOverflowError
+  if isempty(polygons)
+    return nothing
+  end
+  
+  try
+    return Multi(polygons)
+  catch e
+    # Fallback: Return first polygon to prevent crashes (loses Multi structure)
+    @warn "Failed to create Multi Polygon: $e. Returning first polygon only."
+    return polygons[1]
+  end
+end
+
+function parse_wkb_geometrycollection(io::IOBuffer, byte_order::UInt8, has_z::Bool, has_m::Bool, crs_type)
+  read_value = byte_order == 0x01 ? ltoh : ntoh
+  num_geoms = read_value(read(io, UInt32))
+  
+  if num_geoms == 0
+    return nothing
+  end
+  
+  geoms = Geometry[]
+  for i in 1:num_geoms
+    # Each geometry in collection has its own WKB header
+    geom_byte_order = read(io, UInt8)
+    geom_read_value = geom_byte_order == 0x01 ? ltoh : ntoh
+    geom_wkb_type = geom_read_value(read(io, UInt32))
+    
+    geom_base_type = geom_wkb_type & 0x0FFFFFFF
+    geom_has_z = (geom_wkb_type & WKB_Z) != 0
+    geom_has_m = (geom_wkb_type & WKB_M) != 0
+    
+    # Parse the specific geometry type directly to avoid infinite recursion
+    geom = nothing
+    if geom_base_type == WKB_POINT
+      geom = parse_wkb_point(io, geom_byte_order, geom_has_z, geom_has_m, crs_type)
+    elseif geom_base_type == WKB_LINESTRING
+      geom = parse_wkb_linestring(io, geom_byte_order, geom_has_z, geom_has_m, crs_type)
+    elseif geom_base_type == WKB_POLYGON
+      geom = parse_wkb_polygon(io, geom_byte_order, geom_has_z, geom_has_m, crs_type)
+    elseif geom_base_type == WKB_MULTIPOINT
+      geom = parse_wkb_multipoint(io, geom_byte_order, geom_has_z, geom_has_m, crs_type)
+    elseif geom_base_type == WKB_MULTILINESTRING
+      geom = parse_wkb_multilinestring(io, geom_byte_order, geom_has_z, geom_has_m, crs_type)
+    elseif geom_base_type == WKB_MULTIPOLYGON
+      geom = parse_wkb_multipolygon(io, geom_byte_order, geom_has_z, geom_has_m, crs_type)
+    # Note: Don't support nested GeometryCollections to avoid infinite recursion
+    end
+    
+    if !isnothing(geom)
+      push!(geoms, geom)
+    end
+  end
+  
+  return isempty(geoms) ? nothing : GeometrySet(geoms)
+end
+
+
+function parse_gpb(blob::Vector{UInt8}, crs_type)
+  header, offset = parse_gpb_header(blob)
+  geometry = parse_wkb_geometry(blob, offset, crs_type)
+  return geometry, header.srs_id
+end
+
+function get_crs_from_srid(db::SQLite.DB, srid::Int32)
```

Comment:
Please refactor this function and choose a single strategy to extract the CRS type from the file. Avoid try-catch as much as possible because it hides the actual logic and introduces bugs.

Exception handling is perhaps the single most misused feature of any programming language. If you are not interacting with the outside world, nor interacting with unpredictable inputs from end-users, you should *not* use exceptions.